### PR TITLE
UX expressions name-value pairs

### PR DESCRIPTION
**Pull request by Duckers**
Adds support for Name-value pairs.

Requires new Uno stuff with https://github.com/fusetools/uno/pull/1222

This feature enables things like `<Button Clicked="goto('settings', {enabled: true})" />`

This PR contains:
- [x] Changelog
- [x] Documentation
- [x] Tests

### CODE REVIEW

File: Source/Fuse.Common/NameValuePair.uno [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+				if (key != Name) throw new ArgumentException("Object (NameValuePair) does not contain the given key");
+				return Value;
+			}
+		}
+
+		/** Creates an IObject implementation from an `IArray` of `NameValuePair`.
+			The array doesn't need to contain exclusively `NameValuePair` instances. Objects of other
+			types are ignored.
+		*/
+		public static IObject ObjectFromArray(IArray list)
```
Comment (mortoray) [inline Source/Fuse.Common/NameValuePair.uno]: Why is this here? Nothing uses it.

File: Source/Fuse.Reactive.Expressions/NameValuePair.uno [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+		[UXConstructor]
+		public NameValuePair([UXParameter("Name")] string name, [UXParameter("Value")] Expression value) : base(value)
+		{
+			Name = name;
+		}
+
+		protected override object Compute(object value)
+		{
+			return new Fuse.NameValuePair(Name, value);
```
Comment (mortoray) [inline Source/Fuse.Reactive.Expressions/NameValuePair.uno]: Since you create a new object everytime we could make `name` an `Expression` as well, so long as it evaluates to a string.

Comment (Duckers) [inline Source/Fuse.Reactive.Expressions/NameValuePair.uno]: We could, but I figured it would be a rarely used feature and it adds quite a lot of cost. We could add this later if it turns out to be in high demand.

File: Source/Fuse.Common/NameValuePair.uno (hunk 1 above)
Comment (Duckers) [inline Source/Fuse.Common/NameValuePair.uno]: It would be needed pretty soon. Any feature that actualy relies on NameValuePairs will probably want to convert an IArray of those to an IObject.

**Comment by Duckers** [comment]
Needs to be rebased on master after #195 is merged
